src/Halogen/DatePicker/Component/Time.purs

```diff
@@ -0,0 +1,170 @@
+render s = HH.ul [ HP.classes $ pickerClasses s.time ]
+    (f <$> unwrap s.format)
+  where
+  f cmd = HH.li [HP.classes [HH.ClassName "Picker-component"]] $ [renderCommand cmd]
+
+
+renderCommandEnum :: ∀ m. F.Command -> { title :: String , range  :: Range Int } -> HTML m
+renderCommandEnum cmd conf' = let conf = conf'{range = conf'.range} in
+  HH.slot' cpNum cmd
+    (Num.picker Num.intHasNumberInputVal conf) unit
+    (HE.input $ \(NotifyChange n) -> Update $ \t -> n >>= (_ `F.toSetter cmd` t))
+
+renderCommandChoice :: ∀ m a. BoundedEnum a => Show a => F.Command -> { title :: String , values :: NonEmpty Array (Maybe a) } -> HTML m
+renderCommandChoice cmd conf = HH.slot' cpChoice cmd
+    (Choice.picker
+      (Choice.maybeIntHasChoiceInputVal \n -> (toEnum n :: Maybe a) # maybe "" show)
+      (conf{values = conf.values <#> map fromEnum})
+    )
+    unit
+    (HE.input $ \(NotifyChange n) -> Update $ \t -> n >>= (_ `F.toSetter cmd` t))
+
+renderCommand :: ∀ m. F.Command -> HTML m
+renderCommand cmd = case cmd of
+  F.Placeholder str       -> textElement { text: str}
+  F.Meridiem              -> renderCommandChoice cmd { title: "Meridiem", values: upFromIncluding (bottom :: Maybe Meridiem) }
+  F.Hours24               -> renderCommandEnum cmd { title: "Hours", range: (bottomTop :: Range Hour) <#> fromEnum }
+  F.Hours12               -> renderCommandEnum cmd { title: "Hours", range: (bottomTop :: Range Hour12) <#> fromEnum }
+  F.MinutesTwoDigits      -> renderCommandEnum cmd { title: "Minutes", range: (bottomTop :: Range Minute) <#> fromEnum }
+  F.Minutes               -> renderCommandEnum cmd { title: "Minutes", range: (bottomTop :: Range Minute) <#> fromEnum }
+  F.SecondsTwoDigits      -> renderCommandEnum cmd { title: "Seconds", range: (bottomTop :: Range Second) <#> fromEnum }
+  F.Seconds               -> renderCommandEnum cmd { title: "Seconds", range: (bottomTop :: Range Second) <#> fromEnum }
+  F.Milliseconds          -> renderCommandEnum cmd { title: "Milliseconds", range: (bottomTop :: Range Millisecond) <#> fromEnum }
+  F.MillisecondsTwoDigits -> renderCommandEnum cmd { title: "Milliseconds", range: (bottomTop :: Range Millisecond2) <#> fromEnum }
+  F.MillisecondsShort     -> renderCommandEnum cmd { title: "Milliseconds", range: (bottomTop :: Range Millisecond1) <#> fromEnum }
+
+evalTime ∷ ∀ m . TimeQuery ~> DSL m
+evalTime (Update update next) = do
+  s <- H.get
+  nextTime <- map (steper' s.time InvalidTime <<< maybe (Left false) Right) $
+    case s.time of
+      Just (Right time) -> pure $ update time
+      _  -> buildTime
+  H.modify _{ time = nextTime }
+  when (nextTime /= s.time) $ H.raise (NotifyChange nextTime)
+  pure next
+
+
+buildTime :: ∀ m. DSL m (Maybe Time)
+buildTime = do
+  {format} <- H.get
+  mbKleisliEndo <- for (sort $ unwrap format) mkKleisli
+  pure $ map fold (sequence mbKleisliEndo) >>= \(Join (Star f)) -> f bottom
+  where
+  mkKleisli (F.Placeholder _) = pure $ Just $ mempty
```

Comment:
Oh! That's why `Join (Star ...` here, right? 
Could you add type annotation to `mkKleisli` and maybe type synonim for `Join (Star ...` thing?